src/Processor/Consumers/IpaffsConsumer.cs

```
@@ -0,0 +1,54 @@
+using System.Text.Json;
+using Defra.TradeImportsDataApi.Domain.CustomsDeclaration;
+using Defra.TradeImportsDataApi.Domain.Events;
+using Defra.TradeImportsProcessor.Processor.Configuration;
+using Defra.TradeImportsProcessor.Processor.Extensions;
+using Defra.TradeImportsProcessor.Processor.Services;
+using Defra.TradeImportsProcessor.Processor.Utils;
+using Defra.TradeImportsProcessor.Processor.Utils.Logging;
+using Microsoft.Extensions.Options;
+using SlimMessageBus;
+using SlimMessageBus.Host.AmazonSQS;
+
+namespace Defra.TradeImportsProcessor.Processor.Consumers;
+
+public class IpaffsConsumer(IOptions<BtmsOptions> btmsOptions, IIpaffsStrategyFactory ipaffsStrategyFactory)
+    : IConsumer<string>,
+        IConsumerWithContext
+{
+    private string MessageId => Context.GetTransportMessage().MessageId;
+
+    public IConsumerContext Context { get; set; } = null!;
+
+    public async Task OnHandle(string received, CancellationToken cancellationToken)
+    {
+        if (btmsOptions.Value.OperatingMode == OperatingMode.Cutover)
+        {
+            var resourceType = Context.GetResourceType();
+
+            if (resourceType == ResourceEventResourceTypes.CustomsDeclaration)
+            {
+                var message = MessageDeserializer.Deserialize<JsonElement>(
+                    received,
+                    Context.Headers.GetContentEncoding()
+                );
+                var customsDeclaration = message.Deserialize<ResourceEvent<CustomsDeclaration>>();
+
+                if (
```

If the factory goes, it will simplify this logic, as the factory itself is introducing logic re: nullable results that we don't need to handle. 

Just having the available enumerable of strategies will be clearer. If we can find a match then use it, otherwise if we have no match we should throw an exception. Currently the code just consumes and reports nothing so we don't know what is happening.